core-bundle/src/Resources/public/core.js

```diff
@@ -2721,7 +2721,7 @@ var Backend =
 					partElement.setStyle('display', null);
 				}
 				Object.each(values, function(value, key) {
-					inputElements[key].set('value', value.toFixed(15));
+					inputElements[key].set('value', value === '' ? '' : (+value).toFixed(15));
```

Comment:
I would find this much more readable

```suggestion
					inputElements[key].set('value', value === '' ? '' : Number(value).toFixed(15));
```